core/src/main/groovy/noe/tomcat/configure/TomcatConfigurator.groovy

```diff
@@ -367,6 +367,46 @@ class TomcatConfigurator {
     return this
   }
 
+  /**
+   * Removes the specified listener to server.xml
+   * @param listenerFQCN the fully qualified class name of the listener
+   * @return this
+   */
+  TomcatConfigurator removeListener(String listenerFQCN) {
+    String serverXml = 'server.xml'
+    File confFile = retrieveConfFile(serverXml)
+    if (confFile?.exists()) {
+      configVault.push(confFile)
+      Node serverNode = getParsedConfig(confFile)
+      Node listener = serverNode.Listener.find { it.@className.contains(listenerFQCN) }
+      serverNode.remove(listener)
+      printNodeToFile(confFile, serverNode)
+    } else {
+      missingConfigFile(serverXml)
+    }
+    return this
+  }
+
+  /**
+   * Adds the upgrade protocol tag to secure connector
+   * @param listenerFQCN the fully qualified class name of the listener
+   * @return this
+   */
+  TomcatConfigurator upgradeProtocol() {
```

Comment:
Can we have unit test for this method, please?

Reply:
I made the unit test about upgradeProtocol

Reply:
Could you please share the code?

Reply:
Thank you for the test, checking it, I have to realized the following:

- Do we really need this functionality as part of TomcatConfigurator API? It is about setting one attribute of one element. Continuing like this the TomcatConfigurator will be messy soon, right?

Why not use incorporate UpgradeProtocol into existing Connector structures? To make possible smth like
````
TomcatConfigurator tconf = new TomcatConfigurator()
  .httpsConnector(
      new SecureHttpConnectorTomcat()
         .setUpgradeProtocol(new UpgradeProtocol('org.apache.coyote.http2.Http2Protocol')))
  .httpConnector(
      new NonSecureHttpConnectorTomcat()
         .setUpgradeProtocol(new UpgradeProtocol('org.apache.coyote.http2.Http2Protocol')))
// ....
````
and you can add some helpers to have:
````
TomcatConfigurator tconf = new TomcatConfigurator()
  .httpsConnector(
      new SecureHttpConnectorTomcat().upgradeProtocolToHttp2Protocol())
  .httpConnector(
      new NonSecureHttpConnectorTomcat().upgradeProtocolToHttp2Protocol())
// ....
````
Yes, this code is longer than the proposed but it is more semantic and transparent.

Reply:
To be honest my task was about "Extend SecureHttpConnectorTomcat and ConnectorConfiguratorTomcat for additional settings e.g. TomcatConfigurator is not able to upgrade protocol for HTTP2 "

Reply:
I can see your point. So would be better to create upgradeProtocolToHttp2Protocol for both secure and NonSecureHttpConnectors? And call upgrade after setting them, right ?

Reply:
Right, I would prefer that.

The code samples take just like an inspiration you don't have to stick with names exactly and that UpgradeProtocal has constructor like this. This was from the top of my head, you probably find better naming or so.